public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -13,6 +13,7 @@ const DEFAULT_KEYS = ['job', 'instance'];
 const EMPTY_SELECTOR = '{}';
 const HISTORY_ITEM_COUNT = 5;
 const HISTORY_COUNT_CUTOFF = 1000 * 60 * 60 * 24; // 24h
+export const DEFAULT_LOOKUP_METRICS_THRESHOLD = 10000; // number of metrics defining an installation that's too big
```

Comment:
Just to understand this, does this mean that any label query performance will correlate to number of metrics? Naively I would assume that even if there are many metrics, if each metric has low cardinality, then query for the metrics labels will be quick.

Reply:
It's certainly a heuristic, assuming that a lot of metrics will mean a lot of timeseries, which ultimately will have slow response times in dynamic lookups. It's exacerbated by us using the series API, which will return *all* series for a certain subset (e.g., `{cluster="us-east1"}`, this can be HUGE. This PR makes sure it's turned off by default, and only if the number of timeseries is low enough, we enable it.